security-proxy/src/main/java/org/georchestra/security/Proxy.java

```diff
@@ -194,21 +194,37 @@ public HttpUriRequest getRedirect(HttpRequest httpRequest, HttpResponse httpResp
     private Permissions sameDomainPermissions;
     private String proxyPermissionsFile;
 
+    /**
+     * This variable controls the socketTimeout parameter passed to the httpclient
+     * configuration, used for proxified queries. Relying on HttpClient's
+     * documentation
+     * (https://hc.apache.org/httpcomponents-client-4.2.x/tutorial/html/connmgmt.html),
+     * it corresponds to the max time between two consecutive packets, and is
+     * expressed in milliseconds.
+     *
+     */
     private Integer httpClientTimeout = 300000;
 
-    private final static String setCookieHeader = "Set-Cookie";
-    private HttpAsyncClientBuilder httpAsyncClientBuilder;
-
     public void setHttpClientTimeout(Integer timeout) {
         this.httpClientTimeout = timeout;
     }
 
-    public void setPublicUrl(String publicUrl) {
-        this.publicUrl = publicUrl;
+    /**
+     * This variable holds the timeout in minutes before the proxified HTTP requests
+     * will be discarded, throwing a java.util.concurrent.TimeoutException, see
+     * return of executeHttpRequest() method below.
+     */
+    private Integer httpQueryTimeout = 20;
+
+    public void setHttpQueryTimeout(Integer httpQueryTimeout) {
+        this.httpQueryTimeout = httpQueryTimeout;
     }
 
-    public Integer getHttpClientTimeout() {
-        return httpClientTimeout;
+    private final static String setCookieHeader = "Set-Cookie";
+    private HttpAsyncClientBuilder httpAsyncClientBuilder;
+
+    public void setPublicUrl(String publicUrl) {
+        this.publicUrl = publicUrl;
     }
 
     public void init() throws Exception {
```

Comment:
You can add a `@PostConstruct` annotation to avoid `init-method` in `proxy-servlet.xml` see PR #2999